contracts/Transaction.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity 0.4.24;
+
+import "./DataProductInterface.sol";
+import "./Ownable.sol";
+import "./SafeMath.sol";
+import "./TransactionInterface.sol";
+
+
+contract Transaction is Ownable, TransactionInterface {
+    using SafeMath for uint256;
+
+    uint8 constant private minScore = 1;
+    uint8 constant private maxScore = 5;
+
+    address private dataProductAddress;
+    DataProductInterface private dataProduct;
+
+    address public buyerAddress;
+    string public buyerPublicKey;
+    string public buyerMetaHash;
+    uint256 public creationTimeStamp;
+    uint256 public rateDeadline;
+    uint256 public deliveryDeadline;
+    uint256 public price;
+    uint256 public fee;
+    bool public purchased;
```

Comment:
In my opinion this attribute could be deleted. All transactions are purchased so there is no need to store `purchased` flag.

Reply:
yeah IMO as well but I'll keep it for now